core/src/test/java/com/novoda/merlin/service/ConnectivityChangesForwarderTest.java

```diff
@@ -110,6 +110,30 @@ public void givenFetchingWithPing_whenUnsuccessful_thenCallsOnDisconnect() {
     }
 
     @Test
+    public void givenNetworkWasConnected_butNullBindListener_whenNotifyingOfInitialState_thenForwardsNetworkAvailableToListener() {
+        connectivityChangesForwarder = new ConnectivityChangesForwarder(
+                networkStatusRetriever, disconnectCallbackManager, connectCallbackManager, null, endpointPinger
+        );
+        givenNetworkWas(CONNECTED);
+
+        connectivityChangesForwarder.forwardInitialNetworkStatus();
+
+        verify(bindCallbackManager, never()).onMerlinBind(any(NetworkStatus.class));
+    }
+
+    @Test
+    public void givenEndpointIsUnavailable_andNullBindListener_whenNotifyingOfInitialState_thenForwardsNetworkAvailableToListener() {
+        connectivityChangesForwarder = new ConnectivityChangesForwarder(
+                networkStatusRetriever, disconnectCallbackManager, connectCallbackManager, null, endpointPinger
+        );
+        given(networkStatusRetriever.retrieveNetworkStatus()).willReturn(AVAILABLE_NETWORK);
```

Comment:
Ah yes, I can see the confusion 😄 

It's because there is a field that is populated after a network ping has been performed called `lastEndpointPingNetworkStatus`, so I think the tests (I copied these) were named for that. I think this can be changed to make this more explicit though 😄  one sec.